server/schedule/checker/merge_checker.go

```diff
@@ -275,3 +280,25 @@ func checkPeerStore(cluster schedule.Cluster, region, adjacent *core.RegionInfo)
 	}
 	return true
 }
+
+// checkTargetPeerStoreAvailable is used to consider the free space of the target store
+// if target stores don't have enough available space, it returns false
+func checkTargetPeerStoreAvailable(cluster schedule.Cluster, region, target *core.RegionInfo) bool {
+	stores := make(map[uint64]bool)
+	for _, p := range region.GetPeers() {
+		stores[p.GetStoreId()] = true
+	}
+	for _, p := range target.GetPeers() {
+		storeID := p.StoreId
+		if _, ok := stores[storeID]; ok {
+			continue
+		}
+		storeInfo := cluster.GetStore(storeID)
+		availableMiB := storeInfo.GetAvailable() / (1 << 20)
+		// because it is rounded down when the approximateSize of region obtained and converted to Mib, so plus 1 MiB
+		if availableMiB < uint64(region.GetApproximateSize()+1) {
```

Comment:
The available size condition should be loose may 10*region_size. For example, if there are many region that needs merge,  the store will use all available size.

Reply:
> The available size condition should be loose may 10*region_size. For example, if there are many region that needs merge, the store will use all available size. The store available update every 10s.

I will update with MergeScheduleLimit * MaxMergeRegionSize

Reply:
Even if changing to `MergeScheduleLimit * MaxMergeRegionSize`, it's still too small.